### PR TITLE
Use the SHA256 SecretRegistry

### PR DESCRIPTION
**Pull request by pirapira**
`raiden-contracts 0.20.0` replaces Keccak hashlocks with SHA256 hashlocks.  This PR changes the client to use SHA256 hashlocks.

At the same time, the new SecretRegistry doesn't exclude 0x00..00 as a secret.  So we can remove the special treatment of 0x00..00.

In order to make sure nobody launches the client in the production environment or on the mainnet, the client now raises exceptions on these circumstances.

### CODE REVIEW

File: raiden/tests/unit/transfer/mediated_transfer/test_initiatorstate.py [hunk 1]
```diff
@@ -978,9 +978,9 @@ def test_initiator_handle_contract_receive_emptyhash_secret_reveal():
         channelidentifiers_to_channels=setup.channel_map,
         pseudo_random_generator=setup.prng,
     )
-    assert len(iteration.events) == 0
-    # make sure the original lock wasn't moved
-    assert transfer.lock.secrethash in setup.channel.our_state.secrethashes_to_lockedlocks
+    assert len(iteration.events) == 3
```
Comment (rakanalh) [inline raiden/tests/unit/transfer/mediated_transfer/test_initiatorstate.py]: To make the test clear on what it is expecting, it's better to change this to assert for the existence of the specific event you're looking for. You can use: [search_for_item](https://github.com/raiden-network/raiden/blob/develop/raiden/tests/utils/events.py#L67) for this.

**Comment by rakanalh** [comment]
Also, could you please rebase the PR? seems like there are changes here that shouldn't be for `.circleci/config.yml`

File: raiden/constants.py [hunk 2]
```diff
@@ -48,11 +49,12 @@
 EMPTY_BALANCE_HASH = BalanceHash(bytes(32))
 EMPTY_LOCK_HASH = LockHash(bytes(32))
 EMPTY_MESSAGE_HASH = AdditionalHash(bytes(32))
-EMPTY_HASH_KECCAK = keccak(EMPTY_HASH)
 EMPTY_SIGNATURE = Signature(bytes(65))
 EMPTY_MERKLE_ROOT = Locksroot(bytes(32))
 EMPTY_SECRET = Secret(b"")
 EMPTY_SECRETHASH = SecretHash(bytes(32))
+ZERO_SECRET = Secret(bytes(32))
```
[the remark this answers is not in the document]
Comment (pirapira) [reply raiden/constants.py]: `EMPTY_SECRET`'s length is zero.  I needed something new with length 20.

File: raiden/transfer/channel.py [hunk 3]
```diff
@@ -617,15 +612,6 @@ def valid_lockedtransfer_check(
 
             result = (False, msg, None)
 
-        # if the message contains the keccak of the empty hash it will never be
-        # usable onchain https://github.com/raiden-network/raiden/issues/3091
-        elif lock.secrethash == EMPTY_HASH_KECCAK:
```
[the remark this answers is not in the document]
Comment (pirapira) [reply raiden/transfer/channel.py]: > an empty hash will be usable on-chain
Yes.

> what's the reasoning behind that decision?
Interoperability with the hash locks on other blockchains.

File: raiden/tests/unit/transfer/mediated_transfer/test_initiatorstate.py (hunk 1 above)
Comment (pirapira) [reply raiden/tests/unit/transfer/mediated_transfer/test_initiatorstate.py]: I'll do this in this PR.

**Comment by pirapira** [comment]
@rakanalh I'll rebase.  The changes in `.circleci/config.yml` are intentional.

File: raiden/tests/unit/transfer/mediated_transfer/test_initiatorstate.py (hunk 1 above)
Comment (pirapira) [reply raiden/tests/unit/transfer/mediated_transfer/test_initiatorstate.py]: I guess I'm looking for EventUnlockSuccess, so added an assert for it.

File: raiden/constants.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (pirapira) [reply raiden/constants.py]: To make the naming consisntent here, I should first remove
```
EMPTY_SECRET = Secret(b"")
```

All other `EMPTY_*` has 32 bytes.

**Comment by pirapira** [comment]
I have to clean up the empty hash first, before coming back here.

File: .circleci/config.yml [hunk 4]
```diff
@@ -673,15 +673,6 @@ workflows:
...
 
-      - smoketest:
-          name: smoketest-matrix-production-geth-3.7
-          py-version: "3.7"
-          transport-layer: "matrix"
-          environment-type: "production"
-          blockchain-type: "geth"
-          requires:
-            - lint-3.7
-
```
Comment (pirapira) [inline .circleci/config.yml]: Instead of removing this, I should change the contracts version for production.

**Pull request merged**